### PR TITLE
Rate at which ap chests will now match default rate that hearts/chests spawn

### PR DESCRIPTION
**Pull request by ScipioWright**

### CODE REVIEW

**Comment by ScipioWright** [comment]
Already tested and it does work

File: data/archipelago/scripts/ap_heart_replacer.lua [hunk 1]
```diff
@@ -11,17 +11,23 @@ local function APHeartReplacer()
...
-                    EntityAddComponent(ap_chest_id, "VariableStorageComponent",
-                            {
-                                name = "biome_name",
-                                value_string = biome_name,
-                            })
-                    break
+            -- hearts/chests have a 30% chance not to spawn in the base game
+            -- the r given here should be exactly the same as the r given by ap_old_spawn_heart
+            local r = ProceduralRandom(x, y)
+            SetRandomSeed(x, y)
```
Comment (heinermann) [inline data/archipelago/scripts/ap_heart_replacer.lua:17]: I don't think we need to set the seed afterwards

File: data/archipelago/scripts/ap_heart_replacer.lua [hunk 2]
```diff
@@ -11,17 +11,23 @@ local function APHeartReplacer()
...
+                for i = biome_data.first_hc, biome_data.first_hc + 19 do
+                    if Globals.MissingLocationsSet:has_key(i) then
+                        -- spawn the chest, set ap_chest_id equal to its entity ID
+                        local ap_chest_id = EntityLoad("data/archipelago/entities/items/pickup/ap_chest_random.xml", x, y)
+                        has_spawned = true
+                        EntityAddComponent(ap_chest_id, "VariableStorageComponent",
+                                {
+                                    name = "biome_name",
+                                    value_string = biome_name,
+                                })
```
Comment (heinermann) [inline data/archipelago/scripts/ap_heart_replacer.lua:28]: we've got this in `ap_utils.lua`

```lua
addNewInternalVariable(ap_chest_id, "biome_name", "value_string", biome_name);
```

File: data/archipelago/scripts/ap_heart_replacer.lua (hunk 1 above)
Comment (ScipioWright) [reply data/archipelago/scripts/ap_heart_replacer.lua:17]: The original spawn_heart script does it, but agreed it looks dumb. I'll try to do more in depth testing to see if it actually affects anything

**Comment by heinermann** [comment]
Nothing major lmk if you want me to merge as-is

**Comment by ScipioWright** [comment]
Yeah, I'll leave a note for myself to come back and test the random thing and put the util script in, but it can be merged (mostly so I can get another pr in for the pedestals without it acting weird since I didn't branch correctly)

**Pull request merged**